### DISCUSSION TITLE
Custom docker image - conda dependency

### DISCUSSION THREAD

**Discussion by Sharathmk99** [bentoml/BentoML#923] [General] [answered]
Hi Everyone,

While creating custom docker image as a base image for BentoML server, we don't need to have conda dependency.

I got this confirmed from Slack channel.

Thank you!

**Comment by parano** [comment]
Yes, that's correct, BentoML will just ignore the conda dependencies specified in the BentoService class if `conda` is not available in the docker base image. Here is the line of code that does this: https://github.com/bentoml/BentoML/blob/v0.8.3/bentoml/saved_bundle/bentoml-init.sh#L13

Also see my reply in the other thread here: https://github.com/bentoml/BentoML/issues/532#issuecomment-662123726
> conda is optional in users' own custom docker base image if the user does not need to use conda for installing dependencies. From BentoML perspective we do want to provide conda in our default docker base images tho. Besides providing another option to install dependencies, it also helps to ensure the python version of the docker image.

**Comment by parano** [comment, marked as answer]
The real hard requirement for a docker base image for BentoML, is that it has to meet at least one of the two requirements below:

1) `bash`, `conda` - will automatically download the right python version and install pip in this case
2) `bash`, `pip`, `python` - user will need to ensure the python install on docker base image matches the python version used to save the BentoService. And no conda dependencies are specified in the BentoService.

For the sake of consistency and reproducibility, it is always recommended to use docker base image with conda installed. The only case to use a docker base image without conda is that - 1) you can make sure the python version is consistent across your team from development to training and serving, 2) you have built robust CI/CD pipeline to test the docker images created with BentoML, 3) you don't need conda to install dependencies, and 4) you have a good reason to not install conda in your docker base image(security, docker image size, etc)
